ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEventsRangeQueries.scala

```diff
@@ -396,22 +424,29 @@ private[events] object EventsTableFlatEventsRangeQueries {
       val partiesAndTemplatesCondition =
         formatPartiesAndTemplatesWhereClause(
           sqlFunctions,
-          "flat_event_witnesses",
+          "active_cs.flat_event_witnesses",
           partiesAndTemplateIds,
         )
       val filteredWitnesses =
-        sqlFunctions.arrayIntersectionValues("flat_event_witnesses", parties)
+        sqlFunctions.arrayIntersectionValues("active_cs.flat_event_witnesses", parties)
       val submittersInPartiesClause =
-        sqlFunctions.arrayIntersectionWhereClause("submitters", parties)
+        sqlFunctions.arrayIntersectionWhereClause("active_cs.submitters", parties)
       SQL"""select #$selectColumns, #$filteredWitnesses as event_witnesses,
-                   case when #$submittersInPartiesClause then command_id else '' end as command_id
-            from participant_events
-            where create_argument is not null
-                  and event_sequential_id > ${range.startExclusive._2: Long}
-                  and event_sequential_id <= ${range.endInclusive._2: Long}
-                  and (create_consumed_at is null or create_consumed_at > ${range.endInclusive._1: Offset})
+                   case when #$submittersInPartiesClause then active_cs.command_id else '' end as command_id
+            from participant_events as active_cs
+            where active_cs.event_kind = 10 -- create
```

Comment:
Seeing the repetition here, I wonder whether we couldn't keep the queries as is by introducing an appropriate view?

Reply:
Yeah these repetitions are a pain I guess it is due to the resilience of anorm's SQL interpolation for re-use :(
Yes that would be a solution for DRY to do it in SQL, but it is something new in a sense that current code-base is consistently keeping SQL logic in these anorm string literals. Views would break that pattern, next to being a code artifact which is not maintained in code, but in code + in flyway update scripts.
We might want to look into ways how we can reuse anorm literals, or switch to something else here which enables reuse, so we have an alternative to the pure SQL approach. Tradeoffs shall be evaluated.
TLDR: I agree completely with the initiative, and added a TODO for later